command/elevator.py

```diff
@@ -0,0 +1,168 @@
+from commands2 import InstantCommand, ParallelCommandGroup, ConditionalCommand, SequentialCommandGroup, WaitCommand
+from robotpy_toolkit_7407.command import SubsystemCommand, T
+from robotpy_toolkit_7407.utils.units import inch, meters, rev
+from wpimath.geometry import Pose3d, Translation3d, Pose2d, Rotation3d
+import math
+import constants
+from robot_systems import Robot, Sensors
+from subsystem import Elevator
+
+class SetArmPositionRobot(SubsystemCommand[Elevator]):
+
+    def __init__(self, subsystem: T, target: Pose3d):
+        super().__init__(subsystem)
+        self.target: Pose3d = target
```

Comment:
It seems like outside of this function you only need length and angle. I love the use of variables to make each part of the calculation clearer, but you only need to use `self.XYZ` if it's something the object needs to keep track of long term.